pkg/cpumanager/cpu_assignment.go

```diff
@@ -0,0 +1,1124 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpumanager
+
+import (
+	"fmt"
+	"maps"
+	"math"
+	"sort"
+
+	"github.com/go-logr/logr"
+	topology "github.com/kubernetes-sigs/dra-driver-cpu/pkg/cpuinfo"
+	"k8s.io/utils/cpuset"
+)
+
+// NOTE: This file is a copy of https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cpumanager/cpu_assignment.go
+// as of commit https://github.com/kubernetes/kubernetes/commit/fd5b2efa76e44c5ef523cd0711f5ed23eb7e6b1a with minor modifications
+
```

Comment:
This is a  a lot of code copied from Kubernetes core! How will this be maintained? What is the sync strategy going to be? Are there any divergences and have they been documented?

Reply:
That’s a valid concern. We decided to start with the Kubelet code to leverage its maturity and existing test coverage. @ffromani and I briefly discussed about long term strategy, but it wasn't very clear at that point and definitely needs more discussion. We may not necessarily continue keeping in sync with kubelet code, as customizing the implementation to fit the DRA driver's specific use cases (and removing unused logic) could be more beneficial in the long term.

Currently, the divergences from kubelet code is minimal. I can add a comment in the code to document them.

Reply:
Thanks, @swatisehgal. That makes sense. Updating the documentation is long overdue, so I will take that up in a follow-up PR after this one merges. I'll make sure to include the suggestions you outlined above.